.github/workflows/ci.yml

```diff
@@ -211,3 +212,21 @@ jobs:
         with:
           name: output
           path: _output/**
+
+      - name: Publish Artifacts to S3
+        run: make -j2 publish BRANCH_NAME=${GITHUB_REF##*/}
+        if: env.AWS_ACCESS_KEY_ID != ''
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: us-east-1
+
+      - name: Promote Artifacts in S3
+        if: github.ref == 'refs/heads/main' && env.AWS_ACCESS_KEY_ID != ''
```

Comment:
Do we also want to promote the `release-*` branch artifacts in S3?

Reply:
I mostly copied this from `upbound/up`'s definition here:
https://github.com/upbound/up/blob/f342ae4503ebe94bc5e699fdaff4310c90c6fa87/.github/workflows/ci.yml#L279

I also checked a release branch suspecting that in the release branches we will need to change the branch name to reflect the release branch's name. But to my surprise, this does not seem to be the case:
https://github.com/upbound/up/blob/e9d8bddb2eadaee601514821606c4178b812c15a/.github/workflows/ci.yml#L279

There's a similar situation also in the Crossplane repo:
https://github.com/crossplane/crossplane/blob/3eee5e824bb25256beb38d86a7cd970f94e3d3d2/.github/workflows/ci.yml#L420

I did not check explicitly, but I assume the make target handles this. Then if this is the case, I don't know if we really need to state it explicitly.